flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/sink/IcebergFilesCommitterMetrics.java

```diff
@@ -54,12 +60,34 @@ void commitDuration(long commitDurationMs) {
     lastCommitDurationMs.set(commitDurationMs);
   }
 
+  /** This is called upon a successful commit. */
   void updateCommitSummary(CommitSummary stats) {
+    elapsedSecondsSinceLastSuccessfulCommit.refreshLastSuccessfulCommitTime();
     committedDataFilesCount.inc(stats.dataFilesCount());
     committedDataFilesRecordCount.inc(stats.dataFilesRecordCount());
     committedDataFilesByteCount.inc(stats.dataFilesByteCount());
     committedDeleteFilesCount.inc(stats.deleteFilesCount());
     committedDeleteFilesRecordCount.inc(stats.deleteFilesRecordCount());
     committedDeleteFilesByteCount.inc(stats.deleteFilesByteCount());
   }
+
+  private static class ElapsedTimeGauge implements Gauge<Long> {
+    private final TimeUnit reportUnit;
+    private volatile long lastSuccessfulCommitTimeNano;
+
+    ElapsedTimeGauge(TimeUnit timeUnit) {
+      this.reportUnit = timeUnit;
+      this.lastSuccessfulCommitTimeNano = System.nanoTime();
+    }
+
+    void refreshLastSuccessfulCommitTime() {
+      this.lastSuccessfulCommitTimeNano = System.nanoTime();
+    }
+
+    @Override
+    public Long getValue() {
+      return reportUnit.convert(
```

Comment:
It's possible that this metric may polled by the metric reporter before a single checkpoint has completed. Can that edge case be handled? Reporting a huge number is going to be confusing

Reply:
> It's possible that this metric may polled by the metric reporter before a single checkpoint has completed. 

@mas-chen you were referring to the first checkpoint, right? this is not an issue. Note that the report value is a delta value as `System.nanoTime() - lastSuccessfulCommitTimeNano`.

Reply:
No, before the first checkpoint. `lastSuccessfulCommitTimeNano` would be 0 and the delta would be large

Reply:
I missed the constructor init for that variable. Nevermind!